### PR TITLE
richard/WhackAMole

### PR DESCRIPTION
**Pull request by RichardSeverich**

### CODE REVIEW

**Comment by RichardSeverich** [comment]
image

**Review by carledriss — CHANGES_REQUESTED** [review]
Could you increase the code coverage ?
https://codecov.io/gh/MinionsGroup/devfund-h/pull/25/diff?src=pr&el=tree#diff-c3JjL21haW4vamF2YS9vcmcvbWluaW9ucy9kZXZmdW5kL3JpY2hhcmQvd2hhY2thbW9sZS9NYXRyaXhGaWxsLmphdmE=

File: src/main/java/org/minions/devfund/richard/whackamole/MatrixFill.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    private static final char MOLE = 'M';
+    private static final char MOLE_WHACKED = 'W';
+    private int molesLeft;
+    private int score;
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void action(final char[][] moleGrid, final int row, final int column, final StringBuilder myStreamBuilder) {
```
Comment (carledriss) [inline src/main/java/org/minions/devfund/richard/whackamole/MatrixFill.java]: this implementation doesn't make sense

File: src/main/java/org/minions/devfund/richard/whackamole/MatrixActions.java [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+ */
+public interface MatrixActions {
+
+    /**
+     * @param moleGrid        This variable is the array.
+     * @param row             This variable is the row.
+     * @param column          This variable is the column.
+     * @param myStreamBuilder This variable is the Stream Builder.
+     */
+    void action(char[][] moleGrid, int row, int column, StringBuilder myStreamBuilder);
```
Comment (carledriss) [inline src/main/java/org/minions/devfund/richard/whackamole/MatrixActions.java]: According to the implementation 
It is not necessary to send char[][] moleGrid, int row, int column fields
Only it is necessary to send a value char

File: src/test/java/org/minions/devfund/richard/whackamole/WhackAMoleTest.java [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+ * Class.
+ */
+public class WhackAMoleTest {
+
+    private WhackAMole myWhackAMole;
+    private static final int ONE_NE = -1;
+    private static final int ONE = 1;
+    private static final int THREE = 3;
+    private static final int FIVE = 5;
+    private static final String GRID = "* * * * * ";
```
Comment (carledriss) [inline src/test/java/org/minions/devfund/richard/whackamole/WhackAMoleTest.java]: Don't create this constants.
Please create final local variables inside test method.

Comment (RichardSeverich) [reply src/test/java/org/minions/devfund/richard/whackamole/WhackAMoleTest.java]: Ok I will do that.

Comment (RichardSeverich) [reply src/test/java/org/minions/devfund/richard/whackamole/WhackAMoleTest.java]: About private static final String GRID = "* * * * * ";
Intellij tell that we should use as constat.

image

**Review by carledriss — APPROVED** [review]

**Pull request merged**